azure-pipelines.yml

```diff
@@ -180,14 +184,71 @@ jobs:
 
 # Build Correctness Jobs
 
-- job: Correctness_Determinism
+- template: eng/pipelines/evaluate-changed-files.yml
+  parameters:
+    jobName: Determine_Changes
+    vmImageName: ubuntu-latest
+    paths:
+      - subset: compilers
+        include:
+        - src/Compilers/*
+        - src/Dependencies/*
+
+- template: eng/pipelines/build-windows-job.yml
+  parameters:
+    jobName: Correctness_Code_Analysis
+    configuration: Debug
+    queueName: Build.Windows.Amd64.VS2022.Pre.Open
+    restoreArguments: -msbuildEngine dotnet /p:UsingToolPdbConverter=false /p:UsingToolVSSDK=false /p:GenerateSatelliteAssemblies=false
+    buildArguments: -msbuildEngine dotnet -runAnalyzers -warnAsError /p:RoslynEnforceCodeStyle=true /p:UsingToolPdbConverter=false /p:UsingToolVSSDK=false /p:GenerateSatelliteAssemblies=false /p:PublishReadyToRun=false
+
+- job: Correctness_Build_Artifacts
   pool:
     name: NetCore1ESPool-Public
     demands: ImageOverride -equals Build.Windows.Amd64.VS2022.Pre.Open
   timeoutInMinutes: 90
   steps:
     - template: eng/pipelines/checkout-windows-task.yml
 
+    - task: PowerShell@2
+      displayName: Restore
+      inputs:
+        filePath: eng/build.ps1
+        arguments: -configuration Release -prepareMachine -ci -restore -binaryLog
+
+    - task: PowerShell@2
+      displayName: Build
+      inputs:
+        filePath: eng/build.ps1
+        arguments: -configuration Release -prepareMachine -ci -build -pack -publish -sign -binaryLog
+
+    - script: $(Build.SourcesDirectory)\artifacts\bin\BuildBoss\Release\net472\BuildBoss.exe  -r "$(Build.SourcesDirectory)/" -c Release -p Roslyn.sln
+      displayName: Validate Build Artifacts
+
+    - task: PowerShell@2
+      displayName: Generate Syntax Files
+      inputs:
+        filePath: eng/generate-compiler-code.ps1
+        arguments: -test -configuration Release
+        
+    - script: $(Build.SourcesDirectory)\.dotnet\dotnet.exe tool run dotnet-format whitespace  $(Build.SourcesDirectory)\src --folder --include-generated --include $(Build.SourcesDirectory)\src\Compilers\CSharp\Portable\Generated\ $(Build.SourcesDirectory)\src\Compilers\VisualBasic\Portable\Generated\ $(Build.SourcesDirectory)\src\ExpressionEvaluator\VisualBasic\Source\ResultProvider\Generated\ --verify-no-changes
+      displayName: Validate Generated Syntax Files
```

Comment:
As long as we're trying to optimize this leg, should we avoid this step for IDE PRs?

Reply:
Done